### ISSUE TITLE
[BUG]: Error messages lost/swallowed when errors cannot be sent across network

### ISSUE THREAD

**Issue by damonfeldman** [dgraph-io/dgraph#8399]
### What version of Dgraph are you using?

21.03

### Tell us a little more about your go-environment?

Shared clusters, cloud version (as of Oct 2022)

### Have you tried reproducing the issue with the latest release?

No

### What is the hardware spec (RAM, CPU, OS)?

cloud

### What steps will reproduce the bug?

Unsure. The problem is we swallow an error, so don't know the root cause of the failure to report the error.

### Expected behavior and actual result.

Zero logs often report a message like:

… x.go:354] Error while writing: write tcp 10.70.65.193:6080->10.70.71.114:59622: write: broken pipe

This section of the Go code sis an error handling function, and the function (successfully) packages the info into a JSON structure, but then fails to send the JSON representing the error. On failure, the failed send message (“broken pipe” in this case) is logged, but the underlying error and code are not logged, resulting in the info being lost.

That is, we see an error about sending the error somewhere, but we don't see the actual error that caused all this to happen.

We should (IMO) always log error messages and codes, so this fits loosely under "swallowing" an error.

**Comment by damonfeldman** [comment]
Here is the code at x.go:354

```
func SetStatus(w http.ResponseWriter, code, msg string) {
	w.Header().Set("Content-Type", "application/json")
	var qr queryRes
	ext := make(map[string]interface{})
	ext["code"] = code
	qr.Errors = append(qr.Errors, &GqlError{Message: msg, Extensions: ext})
	if js, err := json.Marshal(qr); err == nil {
		if _, err := w.Write(js); err != nil {
			glog.Errorf("Error while writing: %+v", err)
		}
	} else {
		Panic(errors.Errorf("Unable to marshal: %+v", qr))
	}
}
```

**Comment by damonfeldman** [comment]
Engineering has commented this is likely not severe, and is the result of some pod/process being down, affecting only some kinds of metrics gathering. Still, it does clog up some zero logs.

**Comment by damonfeldman** [comment]
This appears to be fixed in v22 (maybe 22.0.2). See: https://github.com/dgraph-io/dgraph/blob/release/v22.0.2/x/x.go. 

Holding off on closing but we should link this to a PR and close.
